src/resolve/remote.rs

```diff
@@ -80,45 +82,53 @@ impl Remote {
 
     fn content_type(&self, response: &Response) -> Option<ContentType> {
         // Get a value content type header if any.
-        let header = response.headers().get(header::CONTENT_TYPE);
-        let header = header.and_then(|value| value.to_str().ok());
-
-        match header {
-            Some("text/markdown") => Some(ContentType::Markdown),
-            Some("image/png") | Some("image/jpeg") => Some(ContentType::Image),
-            Some("application/pdf") => Some(ContentType::Pdf),
+        let header = response.headers().get(header::CONTENT_TYPE)
+            .and_then(|raw| raw.to_str().ok())
+            .and_then(|string| string.parse::<mime::Mime>().ok())?;
+
+        match (header.type_(), header.subtype()) {
+            (mime::TEXT, sub) if sub == "markdown" => Some(ContentType::Markdown),
+            (mime::IMAGE, mime::PNG) | (mime::IMAGE, mime::JPEG) => Some(ContentType::Image),
+            (mime::APPLICATION, sub) if sub == "pdf" => Some(ContentType::Pdf),
```

Comment:
Couldn't you just check against `mime::APPLICATION_PDF` and `mime::IMAGE_*` instead of manually comparing against strings?

Reply:
Coverage of standardized mime types seems to be spotty at best, `text/markdown` should be standard too but I did not find it in the list of constants. The lib does not actually solve comparison well, it's not simply possible to assert that the type matches any `image/*` afaik. See https://github.com/hyperium/mime/issues/73 and https://github.com/hyperium/mime/issues/52#issuecomment-324426832